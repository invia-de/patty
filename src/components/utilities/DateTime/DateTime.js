import React from 'react';
import PropTypes from 'prop-types';

import './DateTime.css';

const WEEK_DAYS = ['So', 'Mo', 'Di', 'Mi', 'Do', 'Fr', 'Sa'];

const withLeadingZero = function(number) {
  return ('' + number).padStart(2, '0');
};

export default function DateTime({ value, format, className }) {
  const dateObject = new Date(value);
  const year = dateObject.getFullYear();
  const month = withLeadingZero(dateObject.getMonth() + 1);
  const date = withLeadingZero(dateObject.getDate());

  format = format.replace('wd', WEEK_DAYS[dateObject.getDay()]);
  format = format.replace('yyyy', year);
  format = format.replace('mm', month);
  format = format.replace('dd', date);

  return (
    <time className={className} dateTime={`${year}-${month}-${date}`}>
      {format}
    </time>
  );
}
DateTime.propTypes = {
  /** add your own styles */
  className: PropTypes.string,
  /** wd: weekday, dd: day of the month with leading zero, mm: day of the month with leading zero, yyyy: full year */
  format: PropTypes.string,
  /** the provided date to render */
  value: PropTypes.string.isRequired
};
DateTime.defaultProps = {
  format: 'wd dd.mm.'
};
