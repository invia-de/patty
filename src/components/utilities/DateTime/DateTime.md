```jsx
<DateTime value="2019-12-01" format="wd: dd.mm.yyyy" /><br />
<DateTime value="2019-12-01" /><br />
<DateTime value="2019-12-01" format="yyyy-mm-dd" />
```
