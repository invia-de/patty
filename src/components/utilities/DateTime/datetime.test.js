import React from 'react';
import DateTime from './DateTime';
import { render } from 'react-testing-library';

test('DateTime renders', () => {
  const { container } = render(
    <DateTime value="2019-12-01" format="wd dd.mm.yyyy" />
  );
  expect(container.firstChild).toMatchSnapshot();
});
