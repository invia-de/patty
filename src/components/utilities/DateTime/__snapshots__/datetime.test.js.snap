// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTime renders 1`] = `
<time
  datetime="2019-12-01"
>
  So 01.12.2019
</time>
`;

exports[`DateTime renders 2`] = `
<time
  datetime="2033-01-01"
>
  Sa 01.01.2033
</time>
`;
